storage/tianmu/core/value.h

```diff
@@ -18,42 +18,87 @@
 #define TIANMU_CORE_VALUE_H_
 #pragma once
 
+#include <boost/variant.hpp>
 #include <string>
 #include <variant>
 
 namespace Tianmu {
 namespace core {
+// std::variant has some bug in gcc8
+#if defined(__GNUC__) && (__GNUC__ == 8)
 class Value final {
  public:
   Value() = default;
-  explicit Value(int64_t i) { var = i; }
+  explicit Value(int64_t i) { var_ = i; }
   ~Value() = default;
 
-  bool HasValue() const { return !std::holds_alternative<std::monostate>(var); }
-  bool IsInt() const { return std::holds_alternative<int64_t>(var); }
-  bool IsDouble() const { return std::holds_alternative<double>(var); }
-  bool IsString() const { return std::holds_alternative<std::string>(var); }
-  bool IsStringView() const { return std::holds_alternative<std::string_view>(var); }
+  bool HasValue() const { return var_.which() != static_cast<int>(ValueType::kValueNull); }
+  bool IsInt() const { return var_.which() == static_cast<int>(ValueType::kValueInt); }
+  bool IsDouble() const { return var_.which() == static_cast<int>(ValueType::kValueDouble); }
+  bool IsString() const { return var_.which() == static_cast<int>(ValueType::kVauleString); }
+  bool IsStringView() const { return var_.which() == static_cast<int>(ValueType::kValueStringView); }
 
-  void SetInt(int64_t v) { var = v; }
-  int64_t &GetInt() { return std::get<int64_t>(var); }
-  const int64_t &GetInt() const { return std::get<int64_t>(var); }
+  void SetInt(int64_t v) { var_ = v; }
+  int64_t &GetInt() { return boost::get<int64_t>(var_); }
+  const int64_t &GetInt() const { return boost::get<int64_t>(var_); }
 
-  void SetDouble(double v) { var = v; }
-  double &GetDouble() { return std::get<double>(var); }
-  const double &GetDouble() const { return std::get<double>(var); }
+  void SetDouble(double v) { var_ = v; }
+  double &GetDouble() { return boost::get<double>(var_); }
+  const double &GetDouble() const { return boost::get<double>(var_); }
 
-  void SetString(char *s, uint n) { var.emplace<std::string>(s, n); }
-  std::string &GetString() { return std::get<std::string>(var); }
-  const std::string &GetString() const { return std::get<std::string>(var); }
+  void SetString(char *s, uint n) { var_ = std::string(s, n); }
+  std::string &GetString() { return boost::get<std::string>(var_); }
+  const std::string &GetString() const { return boost::get<std::string>(var_); }
 
-  void SetStringView(char *s, uint n) { var.emplace<std::string_view>(s, n); }
-  std::string_view &GetStringView() { return std::get<std::string_view>(var); }
-  const std::string_view &GetStringView() const { return std::get<std::string_view>(var); }
+  void SetStringView(char *s, uint n) { var_ = std::string_view(s, n); }
+  std::string_view &GetStringView() { return boost::get<std::string_view>(var_); }
```

Comment:
and try do this 
```
class Value {
xxx
xxx
#if (__GCC__) && 

#else

#endif
xxxx
} ; 
```
is a better one. Now, this impl has a bunch of redundent code.

Reply:
OK, I will optimize this code to reduce redundant code.